core/io/marshalls.cpp

```diff
@@ -151,6 +154,8 @@ Error decode_variant(Variant &r_variant, const uint8_t *p_buffer, int p_len, int
 					(*r_len) += 8;
 				}
 			} else {
+				// Since plain floats will always be encoded as double, this decoding code is
```

Comment:
is this true? Why will be always encode float as double?

Reply:
Because @reduz requested it, no other reason. I can revert this part.